addons/website_sale_stock/models/product_product.py

```diff
@@ -1,19 +1,49 @@
 # -*- coding: utf-8 -*-
 # Part of Odoo. See LICENSE file for full copyright and licensing details.
 
-from odoo import models
+from odoo import models, fields, _
 from odoo.http import request
 
 
 class ProductProduct(models.Model):
     _inherit = 'product.product'
 
+    stock_notification_partner_ids = fields.Many2many('res.partner', string='Back in stock Notifications')
+
+    def has_stock_notification(self, partner_id):
```

Comment:
Also this method should be private (I don't think it's called from an rpc)
```suggestion
    def _has_stock_notification(self, partner_id):
```